core/spec.md

```diff
@@ -235,6 +236,7 @@ specification, MUST be adhere to.
   the `id` of the entity.
 - Constraints:
   - MUST be a non-empty URL
+  - MUST be a read-only attribute from a client perspective
```

Comment:
Geez, my memory is failing me but I think @Clemensv and I chatted about this a while ago and I'm not sure what the result of that chat was.  But, some thoughts:
- in the "file" case self pointing in the file isn't interesting. But, if "self" pointed to the online version (where can I find this "for real") then it is interesting
- but, if the file is meant to be stand-alone (or isn't related to anything online) then this field shouldn't be there

I'm thinking that this section is probably ok as-is since this is just defining the field and not saying whether it's required or not. So, maybe in the definition of each entity that wants to include `self` we should say it's required for API access but optional for docs???